CMakeLists.txt

```diff
@@ -52,6 +51,11 @@ else()
     add_compile_options(-g)
 endif()
 
+option(BUILD_SHARED "Build a shared library and CLI too" OFF)
```

Comment:
This should use [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS), I think?

Reply:
Oh that will simplify things, TIL!